Added support for LinkedList<> type

YPermitin opened the pull request:
#37 Added support for LinkedList<> type + unit tests. Added unit tests for TypeHelper.

lukasz-pyrzyk commented on src/BinaryFormatter/Utils/TypeHelper.cs:
```diff
@@ -24,5 +26,45 @@ public static class TypeHelper
+                var listType = typeof(LinkedList<>);
+                var constructedListType = listType.MakeGenericType(genericTypes);
+                return HasConversionOperator(obj.GetType(), constructedListType);
+            } else
```
i guess else statement is redundant

YPermitin replied:
@lukasz-pyrzyk Agree. FIx in the commit 11163ed

lukasz-pyrzyk commented:
Please update ```readme``` and I'm happy to merge and release 2.1 :)

YPermitin commented:
@lukasz-pyrzyk I did it.

Added info about LinkedList and other to the "collection" part.
Also I added info about not supported types.

Is it ok?
